src/controller/user.controller.ts

```diff
@@ -0,0 +1,19 @@
+import type { UserService } from '../services/user.service.js';
+import  type { CreateUserInput } from '../types/types.js';
+
+export class UserController {
+  private userService: UserService;
+
+  constructor(userService: UserService) {
+    this.userService = userService;
+  }
+
+  async createUser(params: CreateUserInput) {
+    try {
+      return await this.userService.createUser(params);
```

Comment:
Precisa desse await?

Reply:
Nesse caso não , foi mais por bug mental mesmo porque em uma api rest comum é bem comum ter que utilizar async e await a maior parte do tempo, também tinha me esquecido que  o graphql nos resolvers ele meio que  aguarda até que todas as promises sejam resolvidas ou rejeitadas antes de continuar.

Reply:
não entendi direito a relação da necessidade do await com utilizar graphql ou rest

Reply:
Puts acho que não me expressei muito bem, porque na verdade o uso de async e await não depende necessariamente de ser graphQL ou Rest, mas de como vou lidar com operações assincronas.

Nesse caso em específico o principalmente motivo é que já estamos utilizando async e await no resolver, então no resolver por si só já estamos "esperando" a operação ser resolvida, então quando eu utilzei no controller, eu fiz que fosse algo redudante como não to fazendo nenhum tratamento adicional no controller, não tem a necessidade da utilização do async e do await, mas o principal coisa que quis dizer é que mesmo se eu  tirasse os await do controller a lógica ainda funcionaria porque no resolver  as operações assincronas são esperadas para ser resolvidas ou rejeitadas.